js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
+      <div className={ styles.category } key={ name }>
+        <div className={ styles.title }>{ name }</div>
+        <div className={ styles.cards }>
+          <div>{ cards }</div>
+        </div>
       </div>
     );
   }
 
-  renderMenuItem (address) {
-    const balance = this.props.balances
-      ? this.renderBalance(address)
+  renderAccountCard (_account) {
+    const { copied } = this.state;
+    const { address, index = null } = _account;
+
+    const account = this.props.accountsInfo[address];
+    const name = (account && account.name && account.name.toUpperCase()) || address;
+    const { tags = [] } = account && account.meta || {};
+
+    const balance = this.renderBalance(address);
+
+    const onClick = () => {
+      this.handleClick(address);
+    };
+
+    const onFocus = () => {
+      this.setState({ focusedItem: index });
+    };
+
+    const classes = [ styles.account ];
+
+    if (index && copied === index) {
+      classes.push(styles.copied);
+    }
+
+    const addressElements = name !== address
+      ? (
+        <div className={ styles.addressContainer }>
+          <span
+            className={ styles.address }
+            onClick={ this.preventEvent }
+            ref={ `address_${index}` }
+            title={ address }
+          >
+            { address }
+          </span>
+        </div>
+      )
       : null;
 
-    const item = (
-      <div className={ styles.account }>
-        <IdentityIcon
-          className={ styles.image }
-          inline center
-          address={ address } />
-        <IdentityName
-          className={ styles.name }
-          address={ address } />
-        { balance }
+    const setRef = (ref) => {
+      const { tagsRef = [] } = this.nameRefs[address] || {};
+      this.nameRefs[address] = { ref, tagsRef };
+    };
+
+    return (
+      <div
+        key={ address }
+        ref={ `account_${index}` }
+        tabIndex={ 0 }
+        className={ classes.join(' ') }
+        onClick={ onClick }
+        onFocus={ onFocus }
+      >
+        <IdentityIcon address={ address } />
+        <div className={ styles.accountInfo }>
+          <div className={ styles.accountName }>
+            <span ref={ setRef }>{ name }</span>
+          </div>
+
+          { this.renderTags(tags, address) }
+          { addressElements }
+          { balance }
+        </div>
       </div>
     );
+  }
+
+  renderTags (tags = [], address) {
+    if (tags.length === 0) {
+      return null;
+    }
+
+    const setRefs = (tagRef) => {
+      const nameRef = this.nameRefs[address];
+
+      if (!nameRef) {
+        return;
+      }
+
+      nameRef.tagsRef.push(tagRef);
+    };
 
     return (
-      <MenuItem
-        className={ styles.menuItem }
-        key={ address }
-        value={ address }
-        label={ item }>
-        { item }
-      </MenuItem>
+      <Tags tags={ tags } setRefs={ setRefs } />
+    );
+  }
+
+  renderBalance (address) {
```

Comment:
This really all looks like we are doing on the accounts page, so we are duplicating the displays. As per earlier comments, having a component that we can actually use (`AccountCard`?) makes 100% sense.